src/qililab/data_management.py

```diff
@@ -227,17 +223,10 @@ def build_platform(
         >>> platform.name
         galadriel
     """
-    if path is None and runcard is None:
-        raise ValueError("`runcard` argument (str | dict) has not been passed to the `build_platform()` function.")
-    if path is not None:
-        if runcard is not None:
-            raise ValueError("Use only the `runcard` argument, `path` argument is deprecated.")
-        warn(
-            "`path` argument is deprecated and will be removed soon. Use the `runcard` argument instead.",
-            DeprecationWarning,
-            stacklevel=2,
+    if not isinstance(runcard, (str, dict)):
+        raise ValueError(
+            f"Mandatory `runcard` argument in `build_platform()`, is not a supported type: (str, dict), is type: {type(runcard)}."
```

Comment:
I would change the way we say it to something least confusing like "Incorrect type for mandatory argument `runcard` in `build_platform(). Expected (str | dict) but given type: {type(runcard)}`"

Reply:
I agree the sentence, its convoluted at the start! Changing it!

Reply:
Commited! 🚀